src/internal.cc

```diff
@@ -163,18 +163,31 @@ internalHostname(void)
     return host;
 }
 
-int
-internalHostnameIs(const char *arg)
+bool
+internalWithMyHostname(const AnyP::Uri &url, const AnyP::PortCfg &listener)
 {
-    wordlist *w;
-
-    if (0 == strcmp(arg, internalHostname()))
-        return 1;
-
-    for (w = Config.hostnameAliases; w; w = w->next)
-        if (0 == strcmp(arg, w->key))
-            return 1;
+    // different ports is a fast NO
+    if (url.port() != listener.s.port())
+        return false;
+
+    // "localhost" in URL can match wildcard IP, or explicit localhost-IP listeners
+    const bool receivedAtLocalhost = (listener.s.isLocalhost() || listener.s.isAnyAddr());
+    if (receivedAtLocalhost && strcasecmp(url.host(), "localhost") == 0)
+        return true;
+
+    // accel ports with defaultsite= can also match
+    if (listener.defaultsite && strcasecmp(url.host(), listener.defaultsite) == 0)
+        return true;
+
+    // accept visible_hostname, unique_hostname, or machines hostname()
```

Comment:
This is supposed to be documenting what detail/setting(s) the test case is looking for, not how the check is implemented.

Reply:
You have this backwards. This comment as-is documents what this caller is trying to accept.

Calling `internalHostname()` is a de-duplication of the code we would otherwise have to write here to test these three configuration cases against the URL.